### PR TITLE
JSONFormatBear.py: Fix erroneous message

### PR DESCRIPTION
**Pull request by Sidsharik**
This fixes display of incorrect message regardless
of enabling json_sort paramater.

Fixes https://github.com/coala/coala-bears/issues/2503

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/js/JSONFormatBear.py [hunk 1]
```diff
@@ -66,10 +66,18 @@ def run(self, filename, file,
         corrected = tuple(line.rstrip(' \n') + '\n' for line in corrected)
         diff = Diff.from_string_arrays(file, corrected)
 
-        if len(diff) > 0:
+        if len(diff) > 0 and json_sort is True:
```
Comment (abhishalya) [inline bears/js/JSONFormatBear.py]: You can just write `if len(diff) > 0 and json_sort`

File: bears/js/JSONFormatBear.py [hunk 2]
```diff
@@ -66,10 +66,18 @@ def run(self, filename, file,
...
             yield Result(self,
                          'This file can be reformatted by sorting keys and '
                          'following indentation.',
                          affected_code=tuple(d.range(filename)
                                              for d in diff.split_diff()),
                          diffs={filename: diff})
+
+        elif len(diff) > 0 and json_sort is False:
```
Comment (abhishalya) [inline bears/js/JSONFormatBear.py]: Same here, `elif len(diff) > 0 and not json_sort:`

**Comment by li-boxuan** [comment]
Please do not open a new pull request. Fix the first pull request you create. https://github.com/coala/coala-bears/pull/2780

**Comment by li-boxuan** [comment]
@abhishalya If you don't mind can you migrate your comments to the old PR? Sorry for the inconvenience.

**Comment by abhishalya** [comment]
No problem @li-boxuan :smile:

**Pull request closed without merging**